R/child_care_subsidy.R

```diff
@@ -0,0 +1,163 @@
+#' @param annual_cap_subsidy (numeric) Amount at which annual subsidies are capped for those who earn more than `annual_cap_income`. 
+#' 
+#' @param activity_test_1_brackets (numeric vector) The activity levels at which the activity test increases.
+#' @param activity_test_1_hours (numeric vector) The hours corresponding to the step increase in `activity_test_1_brackets`.
+#' 
+#' @param income_test_bracket_1,income_test_bracket_2,income_test_bracket_3,income_test_bracket_4,income_test_bracket_5 (numeric) The steps at which income test 1 changes rates. Note the strange structure \url{https://www.humanservices.gov.au/individuals/services/centrelink/child-care-subsidy/payments/how-your-income-affects-it}.
+#' @param taper_1,taper_2,taper_3 (numeric) The proportion of the hourly cap retained. Note that the rate only decreases between each odd bracket.
+#' 
+#' @return The annual child care subsidy payable per child.
+#' 
+#' @examples
+#' Example from http://guides.dss.gov.au/family-assistance-guide/3/5/4
+#' child_care_subsidy(family_annual_income = 175000,
+#'                    activity_level = 40,
+#'                    activity_exemption = FALSE,
+#'                    child_age = 3,
+#'                    type_of_day_care = "cbdc",
+#'                    cost_hour = 20,
+#'                    hours_day_care_fortnight = 80,
+#'                    early_education_program = FALSE)
+#'                    
+#'@export 
+
+
+
+child_care_subsidy <- function(family_annual_income = 0,
+                               activity_level = Inf,
+                               activity_exemption = FALSE,
+                               child_age = 3,
+                               type_of_day_care = c("cbdc", "oshc", "fdc", "ihc"),
+                               hours_day_care_fortnight,
+                               cost_hour,
+                               early_education_program = FALSE,
+                               
+                               cbdc_hourly_cap = 11.77,
+                               fdc_hourly_cap = 10.90,
+                               oshc_hourly_cap = 10.29,
+                               ihc_hourly_cap = 25.48,
+                               
+                               annual_cap_income = 186958,
+                               annual_cap_subsidy = 10190,
+                               
+                               income_test_bracket_1 = 66958,
+                               income_test_bracket_2 = 171958,
+                               income_test_bracket_3 = 251248,
+                               income_test_bracket_4 = 341248,
+                               income_test_bracket_5 = 354248,
+                               taper_1 = 0.85,
+                               taper_2 = 0.5,
+                               taper_3 = 0.2,
+                               
+                               activity_test_1_brackets = c(0, 8, 16.00001, 48.00001),
```

Comment:
This seems a bit hacky but it essentially makes 16 and 48 not inclusive in the steps